crypto/x509/x509_vfy.c

```diff
@@ -3165,9 +3167,9 @@ static int build_chain(X509_STORE_CTX *ctx)
                         X509_free(curr);
                         ctx->num_untrusted = --num;
                         (void)sk_X509_set(ctx->chain, num, issuer);
-                        curr = issuer;
-                        /* no need to update self_signed */
                     }
+                } else { /* Likely not reachable, but just in case */
+                    X509_free(issuer);
```

Comment:
Yes, and in this case additionally `num != ctx->num_untrusted` holds,
which means that it is not the first self-signed trusted cert we are adding,
and this should never happen!
It look like that also practically does not since I am not aware of any reported mem leak, which inevitably would occur without the new `X509_free(issuer)`.

How about removing the new 
```
                } else { /* Likely not reachable, but just in case */
                    X509_free(issuer);
```
again and replacing instead a little further up
```
                } else if (num == ctx->num_untrusted) {
```
by
```
                } else {
                    /* This should be the first self-signed trusted cert */
                    if (!ossl_assert(ctx->num_untrusted == num)) {
                        X509_free(issuer);
                        goto int_err;
                    }
```

Reply:
Done this now, given @t8m's positive reaction.